### PR TITLE
Remove all Java and JS tools, use Lz4

### PR DESCRIPTION
**Pull request by stijndcl**
This PR adds most of my changes so far to a new branch to play around with. This is also published as a Docker image:

```sh
docker pull ghcr.io/unipept/stijn-thesis:rust-complete
```

### CODE REVIEW

**Comment by pverscha** [comment]
After trying to start the pipeline, I already have one comment :P It would be very nice if you could add write an extra script that automatically builds all required Rust-utilities for the `build_database.sh` script and that automatically moves the compiled binaries to the right directory. This can be created in the `scripts` directory with the name `compile_binaries.sh`. We'll discuss during the next meeting if we should integrate this compilation step in the `build_database.sh` script or not.

**Comment by stijndcl** [comment]
I wanted to do this with a Makefile but then I heard about your background with those :p Definitely open to quickly making one, just keep in mind that compiling binaries requires the Rust tool chain to be installed as well.

**Comment by pverscha** [comment]
@stijndcl No worries about the Rust toolchain. For this reason, I would (at least temporarily) make a separate script for compiling the binaries.

File: scripts/build_database.sh [hunk 1]
```diff
@@ -71,21 +71,19 @@ Dependencies:
...
   * parallel
+  * lz4
 END
 }
 
 # This function removes all temporary files that have been created by this script.
 clean() {
 	# Clean contents of temporary directory
-	rm -rf "$TEMP_DIR/$UNIPEPT_TEMP_CONSTANT"
+#	rm -rf "$TEMP_DIR/$UNIPEPT_TEMP_CONSTANT"
```
Comment (pverscha) [inline scripts/build_database.sh:86]: I think this needs to be uncommented again? Or is there a reason why the temporary files are no longer deleted?

File: scripts/helper_scripts/unipept-database-rs/src/bin/xml-parser.rs [hunk 2]
```diff
@@ -4,6 +4,7 @@ use std::num::NonZeroUsize;
 use anyhow::{Context, Result};
 use clap::Parser;
 use smartstring::{LazyCompact, SmartString};
+use unipept_database::uniprot::UniprotType;
```
Comment (pverscha) [inline scripts/helper_scripts/unipept-database-rs/src/bin/xml-parser.rs:7]: This thing checks whether the Uniprot type is set to either `swissprot` or `trembl`, but these are not the two only options. We can also use this script to process and download so-called "reference proteomes" that can have any ID (mostly in the format U000xxxx, but that's not always the case). So I think we should remove this check here.

**Review by pverscha — CHANGES_REQUESTED** [review]
A few minor comments, but looks very good for the remainder of the code :)

**Comment by pverscha** [comment]
Ok, so I decided to already merge this code (since we are currently going to use it for a sideproject) and then update the branch `feature/stijn-changes` later with the two things that I left as a remark :)

**Comment by pverscha** [comment]
Ok, one more remark. When the pathname of one of the arguments contains spaces, I get the following error: 

```
Compressing 1 of 44 for swissprot
./build_database.sh: line 451: /Volumes/T7.lz4: Operation not permitted
```

(The full pathname was `/Volumes/T7 (II)/stijn/index` for reference)

File: scripts/helper_scripts/unipept-database-rs/src/bin/xml-parser.rs (hunk 2 above)
Comment (stijndcl) [reply scripts/helper_scripts/unipept-database-rs/src/bin/xml-parser.rs:7]: Didn't know this was possible ^^ I agree we should remove it then

**Pull request merged**